src/D2L.CodeStyle.Annotations/Contract/OnlyVisibleToTypeAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace D2L.CodeStyle.Annotations.Contract {
+
+	[AttributeUsage(
+		validOn: AttributeTargets.Method | AttributeTargets.Property,
+		AllowMultiple = true,
+		Inherited = false
+	)]
+	public sealed class OnlyVisibleToTypeAttribute : Attribute {
+
+		public OnlyVisibleToTypeAttribute( string fullyQualifiedTypeName, string assemblyName ) {
+			FullyQualifiedTypeName = fullyQualifiedTypeName;
+			AssemblyName = assemblyName;
+		}

```

Comment:
Could consider this to make `nameof()` usage more ergonomic when the type is in compilation.

```suggestion
		public OnlyVisibleToTypeAttribute( string @namespace, string typeName, string assemblyName ) {
			FullyQualifiedTypeName = $"{@namespace}.{typeName}";
			AssemblyName = assemblyName;
		}
```

Reply:
(nameof being helpful for renames and compiler errors when things get removed so they get cleaned up)

Reply:
I would prefer to leave it in the form `GetTypeByMetadataName` expects it.  This includes container classes & generics.

Reply:
`OnlyVisibleToTypeAttribute( Type )` overload has been added.

Reply:
Do we need the stringly typed one for any case? It hinders "find all references" I think.